addons/survey/static/src/js/survey_form.js

```diff
@@ -289,6 +450,60 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
     // INIT FIELDS TOOLS
     // -------------------------------------------------------------------------
 
+    /**
+     * Will initialize the breadcrumb widget that handles navigation to a previously filled in page.
+     *
+     * @private
+     */
+    _initBreadcrumb: function () {
+        var $breadcrumb = this.$('.o_survey_breadcrumb_container');
+        var pageId = this.$('input[name=page_id]').val();
+        if ($breadcrumb.length) {
+            this.surveyBreadcrumbWidget = new publicWidget.registry.SurveyBreadcrumbWidget(this, {
+                'canGoBack': $breadcrumb.data('canGoBack'),
+                'currentPageId': pageId ? parseInt(pageId) : 0,
+                'pages': $breadcrumb.data('pages'),
+            });
+            this.surveyBreadcrumbWidget.appendTo($breadcrumb);
+            $breadcrumb.removeClass('d-none');  // hidden by default to avoid having ghost div in start screen
+        }
+    },
+
+    /**
+     * Called after survey submit to update the breadcrumb to the right page.
+     * If we have no "page_id", it means we're on the 'end screen', meaning no breadcrumb.
+     */
+    _updateBreadcrumb: function () {
+        if (this.surveyBreadcrumbWidget) {
+            var pageId = this.$('input[name=page_id]').val();
```

Comment:
Isn' it possible to hold a state on the form instead of relying on some (a bit obscure) dom content ?

Like form_is_start, form_is_ended, ...

Reply:
Refactored.